### ISSUE TITLE
FEAT: enhance skrub's TableReport

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#811]
### Is your feature request related to a problem? Please describe.

Following #806, we can improve the TableReport from skrub and have something more thorough in skore.

### Describe the solution you'd like

Have something like:
```python
skore.analyze_table(df, project="my_project", prefix="/my_data")
```
that separately puts in the skore UI (in the skore UI better than output of notebook because it clutters with output graphs and code cells):
- skrub Table Report
    - they choose not to use javascript because they want minimal dependencies, but we can do something nicer
- seaborn pairplot
- what y-data profiling does: percentage of missing values
- in `/my_data`, there is table analysis, correlations, column 1, etc
- compare different datasets

**Comment by rouk1** [comment]
> * he skrub team chose not to use javascript because they want minimal dependencies, but we can do something nicer

FYI Skrub table report actually has [1000+ lines of js](https://github.com/skrub-data/skrub/blob/main/skrub/_reporting/_data/templates/report.js) ; )

**Comment by sylvaincom** [comment]
Could be a step of #834

**Comment by sylvaincom** [comment]
Got this from a user interview:

Make it possible to customize the report table as the data varies depending on the industry, whether it pertains to insurance or travel, the analysis approach differs; in some domains, the average is not relevant for example

**Comment by glemaitre** [comment]
Now, that we have the different report, I think a first step is to compute the `TableReport` via a new accessor. We should make some tests to have a good API there but we could have:

My thoughts is to create a display. Internally, we might want to call the summarize function and not the `TableReport` itself.

Discussing with @Vincent-Maladiere, we thought that the API could be something like:

```python
display = report.data.analyze(data_souce="train"/"test"/"all")
display  # Show the HTML representation of the `TableReport`
display.plot(kind="distribution", feature=...)
```

**Comment by Vincent-Maladiere** [comment]
Below are some additional feature ideas that could be included with the table report display.

- Plot multiple single distributions, which we already have in the TableReport:
	```python
	display.plot(kind="distribution_1d", cols=...)
	```
- Plot pairwise distributions, which is currently not supported in TableReport, between features and the target:
	```python
	display.plot(kind="distribution_2d", cols=...)
	```
- Plotting associations heatmap:
	```python
	display.plot(kind="pearson", cols=...)
	display.plot(kind="cramer", cols=...)
	```
	This would be close to what users would get with a regular `sns.heatmap(X.corr())`, but with our metrics.

Other ideas? WDYT?

**Comment by Vincent-Maladiere** [comment]
Looping back on https://github.com/skrub-data/skrub/issues/1160, the idea of selecting and holding specific fields of the display is interesting

**Comment by glemaitre** [comment]
Indeed, the `plot` method would be a method to check a single visualization into details and the HTML repr is really an full reporting.

**Comment by sylvaincom** [comment]
Another idea: a comparison of table reports could also help to monitor things like data drift

**Comment by glemaitre** [comment]
> Another idea: a comparison of table reports could also help to monitor things like data drift

I think is is a good remark. We have to better define what we want to compare. I think in this case, we should have another display:

```python
report.data.compare(...)
```

It will be a `DatasetComparatorDisplay` in this case. I can foresee different comparison:

- for an `EstimatorReport`, we potentially want to check the distribution between the train and test set.
- for a `CrossValidationReport`, you might want to compare data across fold (`KFold` without shuffling on ordering data can be surprising)
- for an `EstimatorReport` you might want to be able to compare an external set (for instance a set capture during inference) and compare it with the train set.

**Comment by glemaitre** [comment]
Another use case reported by @Vincent-Maladiere is to analyze the distribution of the predictions. It might belong in another accessor that is more towards model inspection.

**Comment by Vincent-Maladiere** [comment]
> for an EstimatorReport you might want to be able to compare an external set (for instance a set capture during inference) and compare it with the train set.

My hunch is that this one would be most helpful for users. Note that it addresses issues at the production/inference monitoring or debugging level

**Comment by sylvaincom** [comment]
Another idea: as the model is a scikit-learn pipeline, we know what is the preprocessing; maybe having a table report before and after the preprocessing would be nice; or even after each transformation

But I think that this is kind of already tackled with skrub expressions?
